src/rust/engine/task_executor/src/lib.rs

```diff
@@ -50,47 +51,20 @@ impl Executor {
     self.handle.enter(f)
   }
 
-  ///
-  /// Drive running of a Future on a tokio Runtime as a new Task.
-  ///
-  /// The future will be driven to completion, but the result can't be accessed directly.
-  ///
-  /// This may be useful e.g. if you want to kick off a potentially long-running task, which will
-  /// notify dependees of its completion over an mpsc channel.
-  ///
-  pub fn spawn_and_ignore<F: Future<Output = ()> + Send + 'static>(&self, future: F) {
-    tokio::spawn(Self::future_with_correct_context(future));
-  }
-
   ///
   /// Run a Future on a tokio Runtime as a new Task, and return a Future handle to it.
   ///
-  /// If the returned Future is dropped, the computation will still continue to completion: see the
-  /// tokio::task::JoinHandle docs.
-  ///
-  /// This may be useful for tokio tasks which use the tokio blocking feature (unrelated to the
-  /// Executor::block_on method). When tokio blocking tasks run, they prevent progress on any
-  /// futures running in the same task. e.g. if you run f1.select(f2) and f1 and f2 are
-  /// tokio blocking futures, f1 and f2 will not run in parallel, defeating the point of select.
+  /// Unlike tokio::spawn, if the background Task panics, the returned Future will too.
   ///
-  /// On the other hand, if you run:
-  /// spawn_oneshot(f1).select(spawn_oneshot(f2))
-  /// those futures will run in parallel.
+  /// If the returned Future is dropped, the computation will still continue to completion: see
+  /// https://docs.rs/tokio/0.2.20/tokio/task/struct.JoinHandle.html
   ///
-  /// Using spawn_oneshot allows for selecting the granularity when using tokio blocking.
-  ///
-  /// See https://docs.rs/tokio-threadpool/0.1.15/tokio_threadpool/fn.blocking.html for details of
-  /// tokio blocking.
-  ///
-  pub async fn spawn_oneshot<O: Send + 'static, F: Future<Output = O> + Send + 'static>(
+  pub fn spawn<O: Send + 'static, F: Future<Output = O> + Send + 'static>(
     &self,
     future: F,
-  ) -> O {
-    // NB: We unwrap here because the only thing that should cause an error in a spawned task is a
-    // panic, in which case we want to propagate that.
+  ) -> impl Future<Output = O> {
     tokio::spawn(Self::future_with_correct_context(future))
-      .await
-      .unwrap()
+      .map(|e| e.expect("Background task exited unsafely."))
```

Comment:
Maybe this is an idiom but naming the `Result` you're `exepect`'ing on here `e` is pretty confusing. It's an `r` and you're expecting it's `Ok` and _not_ `Err` ... so doubly confusing. If not an idiom, maybe `r` or `result` or `join_result`.

Reply:
Thanks, reasonable.